src/devices/FtCommon/FtDevice.cs

```diff
@@ -64,9 +66,23 @@ public FtDevice(FtFlag flags, FtDeviceType type, uint id, uint locId, string ser
         /// Creates I2C bus related to this device
         /// </summary>
         /// <returns>I2cBus instance</returns>
-        public I2cBus CreateI2cBus()
-        {
-            return new Ft4222I2cBus(this);
-        }
+        /// <remarks>You can create either an I2C, either an SPI device.</remarks>
+        public virtual I2cBus CreateI2cBus() => throw new NotImplementedException();
```

Comment:
Make this abstract. Or, if it is reasonable that a derived class won't support it, throw `NotSupportedException` instead. `NotImplementedException` leaves the feeling that something was forgotten.

Reply:
I won't use abstract as this class can be used in real and is needed, so will use virtual. For the exception, yes, will change. Good point.